lib/Lower/Bridge.cpp

```diff
@@ -574,41 +380,328 @@ class FirConverter : public Fortran::lower::AbstractConverter {
+    builder->setInsertionPointToEnd(ifStmtBlock);
+  }
+
+  void genFIR(Fortran::lower::pft::Evaluation &eval,
+              const Fortran::parser::WaitStmt &) {
+    TODO();
+  }
+  void genFIR(Fortran::lower::pft::Evaluation &eval,
+              const Fortran::parser::WhereStmt &) {
+    TODO();
+  }
+  void genFIR(Fortran::lower::pft::Evaluation &eval,
+              const Fortran::parser::ComputedGotoStmt &stmt) {
     auto *exp = Fortran::semantics::GetExpr(
         std::get<Fortran::parser::ScalarIntExpr>(stmt.t));
-    auto e1 = genExprValue(*exp);
+    auto e1{genExprValue(*exp)};
     (void)e1;
     TODO();
   }
-  void genFIR(const Fortran::parser::ForallStmt &) { TODO(); }
-  void genFIR(const Fortran::parser::ArithmeticIfStmt &stmt) {
+  void genFIR(Fortran::lower::pft::Evaluation &eval,
+              const Fortran::parser::ForallStmt &) {
+    TODO();
+  }
+  void genFIR(Fortran::lower::pft::Evaluation &eval,
+              const Fortran::parser::ArithmeticIfStmt &stmt) {
     auto *exp =
         Fortran::semantics::GetExpr(std::get<Fortran::parser::Expr>(stmt.t));
-    auto e1 = genExprValue(*exp);
+    auto e1{genExprValue(*exp)};
     (void)e1;
     TODO();
   }
-  void genFIR(const Fortran::parser::AssignedGotoStmt &) { TODO(); }
+  void genFIR(Fortran::lower::pft::Evaluation &eval,
+              const Fortran::parser::AssignedGotoStmt &) {
+    TODO();
+  }
+
+  void genFIR(Fortran::lower::pft::Evaluation &eval,
+              const Fortran::parser::AssociateConstruct &) {
+    TODO();
+  }
+  void genFIR(Fortran::lower::pft::Evaluation &eval,
+              const Fortran::parser::BlockConstruct &) {
+    TODO();
+  }
+  void genFIR(Fortran::lower::pft::Evaluation &eval,
+              const Fortran::parser::ChangeTeamConstruct &) {
+    TODO();
+  }
+  void genFIR(Fortran::lower::pft::Evaluation &eval,
+              const Fortran::parser::CriticalConstruct &) {
+    TODO();
+  }
+
+  // Information for generating a structured or unstructured increment loop.
+  struct IncrementLoopInfo {
+    IncrementLoopInfo(
+        Fortran::semantics::Symbol *sym,
+        const Fortran::parser::ScalarExpr &lowerExpr,
+        const Fortran::parser::ScalarExpr &upperExpr,
+        const std::optional<Fortran::parser::ScalarExpr> &stepExpr,
+        mlir::Type type, mlir::Block *preheader)
+        : loopVariableSym{sym}, lowerExpr{lowerExpr}, upperExpr{upperExpr},
+          stepExpr{stepExpr}, loopVariableType{type} {}
+
+    bool isStructured() const { return headerBlock == nullptr; }
+
+    // Data members for both structured and unstructured loops.
+    Fortran::semantics::Symbol *loopVariableSym;
+    const Fortran::parser::ScalarExpr &lowerExpr;
+    const Fortran::parser::ScalarExpr &upperExpr;
+    const std::optional<Fortran::parser::ScalarExpr> &stepExpr;
+    mlir::Type loopVariableType;
+    mlir::Value loopVariable{mlir::Value{}};
+    mlir::Value stepValue{mlir::Value{}}; // possible uses in multiple blocks
```

Comment:
`mlir::Value x{};` should be sufficient to make a null value.

Reply:
Thanks